CalibTracker/SiPixelLorentzAngle/src/SiPixelLorentzAnglePCLHarvester.cc

```diff
@@ -351,7 +380,7 @@ void SiPixelLorentzAnglePCLHarvester::dqmEndJob(DQMStore::IBooker& iBooker, DQMS
       // if the fit quality is OK
       if (prob > fitProbCut_) {
```

Comment:
I just want to draw your attention to the fact that right now `fitProbCut_` is set to 0.5

https://github.com/CMSTrackerDPG/cmssw/blob/dc0f55927c5f8cc878e03c1cae6b8cecf87847b4/CalibTracker/SiPixelLorentzAngle/src/SiPixelLorentzAnglePCLHarvester.cc#L498

which is not tuned at all, it's just a random value I picked.

thinking in the long run for the multi-run harvesting applications, a part from the minimum number of events in the MRH configuration, I think it would be good to have some sort of internal logic from within the SiPixel LA harvester to veto the creation of the payload (as it is done for all the other Tracker PCL workflows)
At the moment it just checks that the single "module-group" fit probability is higher than this given cut and if that condition is not satisfied it copies the values from the current value stored in DB (not stopping the payload creation).
If a more refined logic is introduced the input can be balanced in a more refined way, since the MRH will retry aggregating more statistics until a valid payload is produced.

@wweiphy @tvami @ferencek @OzAmram

Reply:
We have a similar issue with the SiPixelQuality payloads where the old payloads gets appended when the occupancy threshold is not passed. So it looks like we need a similar fix for both types of payloads.

Concerning the `fitProbCut`, I guess it would make sense to have a look at the distribution and see if some other cut value would make more sense. However, the question of a more refined internal logic still remains. Once the required minimum number of events is reached, is it possible to continue collecting statistics until some other conditions is met (and only then the payload is created). I really don't know much about MRH and how complicated the logic can be.

Reply:
> However, the question of a more refined internal logic still remains. Once the required minimum number of events is reached, is it possible to continue collecting statistics until some other conditions is met (and only then the payload is created). I really don't know much about MRH and how complicated the logic can be.

the logic better be implemented on the cmssw side, where you have full control and NOT in the MRH backend.